pkg/kubecost/allocation.go

```diff
@@ -2260,8 +2260,39 @@ func (asr *AllocationSetRange) Accumulate() (*AllocationSet, error) {
 	return allocSet, nil
 }
 
-// TODO accumulate into lower-resolution chunks of the given resolution
-// func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) *AllocationSetRange
+// AccumulateBy sums AllocationSets based on the resolution given. The resolution given is subject to the scale used for the AllocationSets.
+// If the requested resolution is smaller than the window of an AllocationSet then the resolution will default to the duration of a set.
+// Resolutions larger than the duration of the entire AllocationSetRange will default to the duration of the range.
+func (asr *AllocationSetRange) AccumulateBy(resolution time.Duration) (*AllocationSetRange, error) {
+	allocSetRange := &AllocationSetRange{allocations: []*AllocationSet{}}
+	var allocSet *AllocationSet
+	var currAccumulatedSum time.Duration
+	var err error
+
+	asr.Lock()
+	defer asr.Unlock()
+
+	for i, as := range asr.allocations {
+		allocSet, err = allocSet.accumulate(as)
+		if err != nil {
+			return nil, err
+		}
+
+		if allocSet != nil {
+			currAccumulatedSum += allocSet.Window.Duration()
+
+			// check if end of asr to sum the final set
+			// If accumulated sum !>= resolution return 1 accumulated set
```

Comment:
Maybe I'm misreading, but what would `!>=` mean here? (Or maybe it's just supposed to be `>=`?)

Reply:
Just making sure I'm with you here. These are the conditions for closing the current set and opening a new one:
* We've accumulated at least the given resolution. More is also fine -- meaning that even divisibility is not a requirement here, and we round up. (That would be a great detail to document!)
* We've reached the end of the range, so even if we're not at a set of full resolution, close it.

That square with your specification?

Reply:
Fixed the typo on the comment and yep that sums it up pretty well!

The round up will be specific to the resolution of the AllocationSet window durations returned. So essentially will accumulate to the nearest ceiling of the requested resolution. 

E.g. User requests 2.5h resolution and each AllocationSet has a 1h window. The first accumulated set added to the AllocationSetRange will be the sum of the first three set's  for a total window duration of 3h